server/services/stadiums.ts

```diff
@@ -109,3 +110,145 @@ export const getStadiumProfileService = async (
     data: stadium,
   };
 };
+
+export const UpdateStadiumDataService = async (
+  req: Request,
+): Promise<{
+  status: number;
+  data: string | { message: string; stadiumResult: number; UserResult: number };
+}> => {
+  // const { StadiumId } = req.UserData this will happen after create protected routes
+  const StadiumId = 5; //and this will removed
+  const { body } = req;
+
+  const isStadiumExist = await Stadium.findOne({
+    where: { UserId: StadiumId },
+  });
+  if (!isStadiumExist) {
+    return {
+      status: 401,
+      data: 'هذا الملعب غير متاح',
+    };
+  }
+  await updatedValueSchema.validateAsync(body);
+
+  const stadiumResult = await Stadium.update(body, {
+    where: { UserId: StadiumId },
+  });
+
+  const UserResult = await User.update(body, { where: { id: StadiumId } });
+
+  return {
+    status: 200,
+    data: {
+      message: 'تم الحفظ بنجاح',
+      stadiumResult: [stadiumResult as Array<number>][0][0],
+      UserResult: [UserResult as Array<number>][0][0],
+    },
+  };
+};
+
+export const UpdateStadiumGalleryService = async (
+  req: Request,
+): Promise<{
+  status: number;
+  data: string | object;
+}> => {
+  const {
+    body: { image, id, StadiumId },
+  } = req;
+
+  // const { StadiumId } = req.UserData this will happen after create protected routes
+  const checkAuthId = 5; //and this will removed
+
+  if (StadiumId !== checkAuthId) {
+    return {
+      status: 401,
+      data: 'UnAuthorize',
+    };
+  }
+
+  const check = await Gallery.findAll({ where: { id, StadiumId } });
+  if (!check.length) {
+    return {
+      status: 401,
+      data: ' هذه الصورة غير موجودة',
+    };
+  }
+
+  const galleries = await Gallery.update(
+    { image },
+    { where: { id, StadiumId }, returning: true },
+  );
+
+  return {
+    status: 200,
+    data: galleries,
```

Comment:
galleries consist of two things, the number of updated rows and the updated values itself, and here we just need updated values itself